src/js/core.js

```diff
@@ -0,0 +1,126 @@
+var d3 = require('d3');
+var _ = require('underscore');
+var utils = require('./utils');
+
+function compute_svg_width(rect_width, rect_padding, row_length) {
+  return (rect_width + rect_padding) * row_length;
+}
+
+var core = function() {
+  var config = { row_height: 15 };
+  var container_width = 100;
+  var element_padding = 1;
+  var element_width = 1;
+  var labels = [];
+  var rows = undefined;
+  var svg_height = 95;
+
+  // styles, appends, does all the right stuff to the container
+  // so that we can go on to work with the inner <svg>.
+  function oncoprint_container_to_svg(container) {
+    container.style('width', container_width + "px")
+    .style('display', 'inline-block')
+    .style('overflow-x', 'auto')
+    .style('overflow-y', 'hidden');
+
+    return container.append('svg')
+    .attr('width', compute_svg_width(element_width, element_padding, rows[0].length))
+    .attr('height', config.row_height * rows.length);
+  }
+
+  var me = function(container) {
+
```

Comment:
I think it might be nicer to have the rendering a bit more modularized, instead of laundry list of commands. What do you think?